aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,732 @@
+
+The following Razor markup:
+
+.. code-block:: html
+
+  @Html.Raw("<span>Hello World</span>")
+
+Renders this HTML:
+
+.. code-block:: html
+
+  <span>Hello World</span>
+
+Which the browser renders as:
+``Hello World``
+
+.. _razor-code-blocks-label:
+
+Razor code blocks
+------------------
+
+Razor code blocks start with ``@`` and are enclosed by ``{}``. Unlike expressions, C# code inside code blocks is not rendered.
+
+.. comment: sample shows var defined in first code block available in 2nd block. Some customer confusion on that.
+
+.. literalinclude:: razor/sample/Views/Home/Contact.cshtml
+  :language: html
+  :start-after: }
+  :end-before: @* End of greeting *@
+
+Renders this HTML markup:
+
+.. code-block:: html
+
+  <!-- Single statement blocks.  -->
+
+  <!-- Multi-statement block.  -->
+      <p>The greeting is :<br /> Welcome! Today is: Tuesday -Leap year: True</p>
+      <p>The value of your account is: 7 </p>
+
+.. _implicit-transitions-label:
+  
+Implicit transitions
+^^^^^^^^^^^^^^^^^^^^^
+
+The default language in a code block is C#, but you can transition back to HTML. HTML within a code block will transition back into rendering HTML:
+
+.. code-block:: none
+
+  @{
+      var inCSharp = true;
+      <p>Now in HTML, was in C# @inCSharp</p>
+  }
+
+.. _explicit-delimited-transition-label:
+
+Explicit delimited transition
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To define a sub-section of a code block that should render HTML, surround the characters to be rendered with the ``<text>`` tag:
+
+.. code-block:: none
+  :emphasize-lines: 4
+
+  @for (var i = 0; i < people.Length; i++)
+  {
+      var person = people[i];
+      <text>Name: @person.Name</text>
+  }
+
+You generally use this approach when you want to render HTML that is not surrounded by an HTML tag. Without HTML tags, you get a Razor runtime error:
+
+.. code-block:: none
+  :emphasize-lines: 4
+
+  @for (var i = 0; i < people.Length; i++)
+  {
+      var person = people[i];
+      Name: @person.Name
+  }
+
+.. _explicit-line-transition-with-label:
+
+Explicit Line Transition with ``@:``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Typically, HTML tags provide a boundary for Razor to transition into C#:
+
+.. code-block:: none
+  :emphasize-lines: 4
+
+  @for (var i = 0; i < people.Length; i++)
+  {
+      var person = people[i];
+      <text>Name: @person.Name</text>
+  }
+
+To render the Razor above without HTML tags, use the ``@:`` characters:
+
+.. code-block:: none
+  :emphasize-lines: 4
+
+  @for (var i = 0; i < people.Length; i++)
+  {
+      var person = people[i];
+      @:Name: @person.Name
+  }
+
+Without the ``@:`` in the code above, you'd get a Razor run time error.
+
+.. _explicit-email-override-label:
+
+``@()`` explicit expression to override email @ symbol
```

Comment:
I think you should remove this whole section. It's not covering any particular Razor feature and feels a bit random.